skycatalogs/sso_catalog_creator.py

```diff
@@ -0,0 +1,291 @@
+import os
+import sys
+from multiprocessing import Process, Pipe
+import numpy as np
+import sqlite3
+import pandas as pd
+import pyarrow as pa
+import pyarrow.parquet as pq
+from .objects.base_object import LSST_BANDS
+
+"""
+Code for creating sky catalogs for sso objects
+"""
+
+__all__ = ['SsoCatalogCreator']
+
+_DEFAULT_ROW_GROUP_SIZE = 100000      # Maybe could be larger
+
+
+def _do_sso_flux_chunk(send_conn, sso_collection, instrument_needed,
+                       l_bnd, u_bnd):
+    '''
+    end_conn         output connection
+    star_collection  information from main file
+    instrument_needed List of which calculations should be done
+    l_bnd, u_bnd     demarcates slice to process
+
+    returns
+                    dict with keys id, lsst_flux_u, ... lsst_flux_y
+    '''
+    out_dict = {}
+
+    o_list = sso_collection[l_bnd: u_bnd]
+    # out_dict['id'] = [o.get_native_attribute('id') for o in o_list]
+    # out_dict['mjd'] = [o.get_native_attribute('mjd') for o in o_list]
+    out_dict['id'] = list(sso_collection._id[l_bnd: u_bnd])
+    out_dict['mjd'] = list(sso_collection._mjds[l_bnd: u_bnd])
+    if 'lsst' in instrument_needed:
+        all_fluxes = [o.get_LSST_fluxes(as_dict=False, mjd=o._mjd) for o in o_list]
+        all_fluxes_transpose = zip(*all_fluxes)
+        for i, band in enumerate(LSST_BANDS):
+            v = all_fluxes_transpose.__next__()
+            out_dict[f'lsst_flux_{band}'] = v
+
+    # if 'roman' in instrument_needed:
+    #     all_fluxes = [o.get_roman_fluxes(as_dict=False) for o in o_list]
+    #     all_fluxes_transpose = zip(*all_fluxes)
+    #     for i, band in enumerate(ROMAN_BANDS):
+    #         v = all_fluxes_transpose.__next__()
+    #         out_dict[f'roman_flux_{band}'] = v
+
+    if send_conn:
+        send_conn.send(out_dict)
+    else:
+        return out_dict
+
+
+class SsoCatalogCreator:
+    _sso_truth = '/sdf/home/j/jrb/rubin-user/sso/input/20feb2024'
+    _sso_sed = '/sdf/home/j/jrb/rubin-user/sso/sed/solar_sed.db'
```

Comment:
They can be specified when you invoke the creator (which can take an options file). There are options --sso-sed and --sso-truth. Those hard-coded values are just defaults.

Reply:
Since they are supposed to be default values, it seems like it would better to make these the default values in the `create_sc.py` script rather than putting them here.